datafusion/src/sql/utils.rs

```diff
@@ -390,6 +390,42 @@ pub(crate) fn extract_aliases(exprs: &[Expr]) -> HashMap<String, Expr> {
         .collect::<HashMap<String, Expr>>()
 }
 
+/// Returns mapping of each position (`String`) to the expression (`Expr`) it is
+/// aliasing.
+pub(crate) fn extract_positions(exprs: &[Expr]) -> HashMap<String, Expr> {
+    let mut position = 0;
+    exprs
+        .iter()
```

Comment:
You could use the `enumerate` function here to generate the `position` values.
https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.enumerate